src/relax/transform/simplify_norm.cc

```diff
@@ -101,30 +152,37 @@ class NormInferenceSimplifier : public ExprMutator {
       // Instead, we will store the unpacked value in the batch_norm_map_, and replace it
       // at the TupleGetItemNode. And the original batch_norm call will be removed in the
       // follow-up pass `RemoveAllUnused`
-      batch_norm_map_.Set(binding->var, SimplifyBatchNorm(val));
+      if (mode_ == kModeEval) {
+        batch_norm_map_.Set(binding->var, {SimplifyBatchNormInference(val), NullOpt, NullOpt});
+      } else {
+        ICHECK_EQ(mode_, kModeTraining);
+        batch_norm_map_.Set(binding->var, SimplifyBatchNormTraining(val));
+      }
     }
   }
 
  private:
   /*! \brief The mapping from binding var of batch_norm to the unpacked value. */
-  Map<Expr, Expr> batch_norm_map_;
+  Map<Expr, Array<Optional<Expr>>> batch_norm_map_;
+
+  String mode_;
 };
 
 namespace transform {
-Pass SimplifyNormInference() {
+Pass SimplifyNorm(String mode) {
   runtime::TypedPackedFunc<Function(Function, IRModule, PassContext)> pass_func =
       [=](Function f, IRModule m, PassContext pc) {
-        f = Downcast<Function>(NormInferenceSimplifier::Simplify(f));
+        f = Downcast<Function>(NormSimplifier::Simplify(f, mode));
         // Remove original batch_norm op if it's not used.
         return RemoveAllUnused(f);
       };
-  return CreateFunctionPass(/*pass_function=*/pass_func,            //
-                            /*opt_level=*/0,                        //
-                            /*pass_name=*/"SimplifyNormInference",  //
+  return CreateFunctionPass(/*pass_function=*/pass_func,   //
```

Comment:
Sometimes we only need to modify one function in a irmodule. Shall we change it into ModulePass and pass a Optional<String> parameter that specifies the function to simplify, and if it's null, we will simplify all functions in the given IRModule?